src/OpenTelemetry.Instrumentation.StartupHook/StartupHook.cs

```diff
@@ -14,6 +16,17 @@ internal class StartupHook
     /// </summary>
     public static void Initialize()
     {
+        var applicationName = GetApplicationName();
+        StartupHookEventSource.Log.Trace($"StartupHook loaded for application with name {applicationName}.");
+
+        if (IsApplicationInExcludeList(applicationName))
+        {
+            StartupHookEventSource.Log.Trace("Application is in the exclusion list. Skipping initialization.");
```

Comment:
Instead of failing silently should we consider throwing an exception?

Reply:
I don't necessarily see this path as a failure case. Depending on how the application is started, I expect this path to get triggered and purposely not monitor that "application". For example, if the application is started with the command `dotnet run` instead of `dotnet myapp.dll`, two processes are ultimately started: 1. The `dotnet` process with an `AppDomain` of `dotnet`, and 2. the `myapp` process with an `AppDomain` of `myapp`. When `dotnet myapp.dll` is used, one `dotnet` process is started, but the `AppDomain` is `myapp`. When `dotnet run` is used to start the application we don't want to instrument the `dotnet` process/application.

One of the problems that we were seeing was that if we tried to enable the `AspNet` instrumentation library, and started the AspNet Core application using `dotnet run`, the startup hook ultimately attempted to trigger the loading of the `AspNet` instrumentation within the `dotnet` `AppDomain`. The AspNet Core dependencies were not available for that process because it is not an AspNet Core app, so an exception was thrown when we attempted to load the `AspNet` instrumentation. That exception also prevents the starting of the `myapp` process, which would have also attempted to load the startup hook.

I do think that we should provide some documentation on how to see the StartupHook log messages so that these types of startup issues can be easier to diagnose in the future.